src/Microsoft.ML.OnnxConverter/OnnxUtils.cs

```diff
@@ -376,7 +376,7 @@ public ModelArgs(string name, TensorProto.Types.DataType dataType, List<long> di
                 }
             }
             //batch size.
-            dimsLocal?.Insert(0, 1);
```

Comment:
Have you tested that this works? That is, can you take a model exported from ML.NET with this fix and run it with batch size > 1 outside of ML.NET?

Reply:
We should also explore whether ML.NET can run the models with batch sizes > 1

---
In reply to: [375025535](https://github.com/dotnet/machinelearning/pull/4783#discussion_r375025535) [](ancestors = 375025535)

Reply:
As discussed offline, I have taken an ML.NET model and ran it using the onnxruntime python bindings with a batch of 150 rows and streaming it 1 row at a time. Both work as expected.